aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/PersistentMutationOutbox.java

```diff
@@ -58,31 +60,88 @@ final class PersistentMutationOutbox implements MutationOutbox {
     private static final Logger LOG = Amplify.Logging.forNamespace("amplify:aws-datastore");
 
     private final LocalStorageAdapter storage;
-    private final MutationQueue mutationQueue;
     private final Set<TimeBasedUuid> inFlightMutations;
     private final PendingMutation.Converter converter;
     private final Subject<OutboxEvent> events;
     private final Semaphore semaphore;
+    private PendingMutation<? extends Model> loadedMutation;
+    private int numMutationsInStorage;
```

Comment:
Looks like this is only accessed/mutated when the semaphore is acquired - should be okay.